osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -188,6 +189,8 @@ public Dictionary<HitResult, int> MaximumStatistics
 
         public IReadOnlyDictionary<HitResult, int> Statistics => ScoreResultCounts;
 
+        public virtual ILegacyScoreProcessor? CreateLegacyScoreProcessor() => null;
```

Comment:
this virtual still shouldn't be here, should be in `Ruleset` at the least. best case scenario would be having all of this firmly confined within the osu! ruleset project but I am not going to insist on it personally because it's likely to not be very straightforward to achieve and I don't want to send you down that detour without the rest of the team sounding off on the entirety of this

Reply:
moved to `Ruleset`